### PR TITLE
Enables auth sample app to use multiple Firebase Apps.

### PR DESCRIPTION
**Pull request by XiangtianDai**
Note that this reveals #161 .

### CODE REVIEW

File: AuthSamples/Sample/AppManager.h [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    @brief Retrieves the app at the given index.
+    @param index The index of the app to be retrieved, 0 being the default app.
+    @return The app at the given index.
+ */
+- (nullable FIRApp *)appAtIndex:(int)index;
+
+/** @fn recreateAppAtIndex:withOptions:completion:
+    @brief Deletes the app at the given index, and optionally creates it again with given options.
+    @param index The index of the app to be recreated, 0 being the default app.
+    @param options The new options to create the app with, or nil for not to create the app.
```
Comment (protocol86) [inline AuthSamples/Sample/AppManager.h]: maybe: "Optionally, the new options with which app should be created."

File: AuthSamples/Sample/SettingsViewController.m [hunk 2]
```diff
@@ -221,49 +228,68 @@ - (void)toggleAPIHostWithRequestClassName:(NSString *)requestClassName {
...
+}
+
+/** @fn toggleActiveApp
+    @brief Toggles the active Firebase app for the rest of the application.
  */
-- (NSString *)currentProjectID {
-  NSString *APIKey = [FIRApp defaultApp].options.APIKey;
+- (void)toggleActiveApp {
+  AppManager *apps = [AppManager sharedInstance];
+  apps.active = (apps.active + 1) % apps.count;
```
Comment (protocol86) [inline AuthSamples/Sample/SettingsViewController.m]: I am not sure what we are doing here, should these be switched.

File: AuthSamples/Sample/AppManager.h (hunk 1 above)
Comment (XiangtianDai) [inline AuthSamples/Sample/AppManager.h]: Done.

File: AuthSamples/Sample/SettingsViewController.m (hunk 2 above)
Comment (XiangtianDai) [inline AuthSamples/Sample/SettingsViewController.m]: Adds a piece of comment there. Is it better?

**Pull request merged**